src/protean/impl/repository/dict_repo.py

```diff
@@ -49,28 +50,72 @@ def _create_object(self, model_obj):
 
         return model_obj
 
-    def _filter_objects(self, page: int = 1, per_page: int = 10,
-                        order_by: list = (), _excludes=None, **filters):
-        """ Read the repository and return results as per the filer"""
-
-        # Filter the dictionary objects based on the filters
-        items = []
-        excludes = _excludes if _excludes else {}
-        for item in self.conn['data'][self.model_name].values():
+    def _extract_lookup(self, key):
+        """Extract lookup method based on key name format"""
+        parts = key.split('__')
+        # 'exact' is the default lookup if there was no explicit comparison op in `key`
+        #   Assume there is only one `__` in the key.
+        #   FIXME Change for child attribute query support
+        op = 'exact' if len(parts) == 1 else parts[1]
+
+        # Construct and assign the lookup class as a filter criteria
+        return (parts[0], self.get_lookup(op))
+
+    def _extract_confirming_values(self, key, value, negated, db):
```

Comment:
This could be renamed as `_evaluate_lookup`

Reply:
Renamed. Good one. After some time with a PR, these things tend to become transparent 😄